datumaro/util/os_util.py

```diff
@@ -49,6 +66,57 @@ def walk(path, max_depth=None):
 
         yield dirpath, dirnames, filenames
 
+def copytree(src, dst):
+    # Shutil works very slow pre 3.8
+    # https://docs.python.org/3/library/shutil.html#platform-dependent-efficient-copy-operations
+    # https://bugs.python.org/issue33671
+
+    if sys.version_info[1] >= 8:
+        shutil.copytree(src, dst)
+        return
+
+    basedir = osp.dirname(dst)
+    if basedir:
+        os.makedirs(basedir, exist_ok=True)
+
+    if sys.platform == 'windows':
+        # Ignore
+        #   B603: subprocess_without_shell_equals_true
+        #   B607: start_process_with_partial_path
+        # In this case we control what is called and command arguments
+        # PATH overriding is considered low risk
+        subprocess.check_call(["xcopy", src, dst, "/s", "/e"]) # nosec
+    elif sys.platform == 'linux':
+        # As above
+        subprocess.check_call(["cp", "-r", "--reflink=auto", src, dst]) # nosec
```

Comment:
Reflinks are unlikely to be used at all, because only few filesystems support them, and ext3 and 4 do not. The only exception is btrfs (or overlayfs), which are used in docker, but I don't think it is a problem.

Reply:
OK, but please add the double dash argument, like in my suggestion.

Reply:
Added